build/Nuget/Microsoft.NET.Nuget.csproj

```diff
@@ -22,32 +21,29 @@
     <PackageAssets Include="$(OutDir)**\*.*" Exclude="*.log" />
   </ItemGroup>
 
-  <Target Name="Build"
+    <!-- This project is being used as a wrapper project for packing with the nuspec.
+         https://github.com/NuGet/Home/wiki/Adding-nuget-pack-as-a-msbuild-target#packing-using-a-nuspec
+    -->
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
+    <NuspecFile>Microsoft.NET.Sdk.nuspec</NuspecFile>
+    <NuspecBasePath>$(RootOutDir.TrimEnd('\'))</NuspecBasePath>
+    <NuspecProperties>version=$(Version)</NuspecProperties>
+  </PropertyGroup>  
+  
+  <Import Sdk="Microsoft.NET.Sdk" Project="Sdk.targets" />
+
+  <Target Name="Build" 
           DependsOnTargets="LogOutput;
                             InvalidateNuGetCache;
-                            @(NuSpec->'%(LayoutTargetName)')"
+                            @(NuSpec->'%(LayoutTargetName)')" 
           Inputs="@(NuSpec);@(PackageAssets)"
           Outputs="@(NuSpec->'%(NuPkgOutput)')">
-
-    <PropertyGroup>
-      <NuGetExeVersion>3.4.4</NuGetExeVersion>
-      <NuGetTool>$(RepositoryRootDirectory).nuget\$(NuGetExeVersion)\nuget.exe</NuGetTool>
-      <NuGetDownloadUrl>https://dist.nuget.org/win-x86-commandline/v$(NuGetExeVersion)/NuGet.exe</NuGetDownloadUrl>
-      <NuGetDownloadCommand>powershell -noprofile -nologo -command "(new-object System.Net.WebClient).DownloadFile('$(NugetDownloadURL)', '$(NuGetTool)')"</NuGetDownloadCommand>
-    </PropertyGroup>
-
-    <MakeDir Directories="$(NuGetOutDir)" />
-
-    <MakeDir Directories="$(RepositoryRootDirectory).nuget\$(NuGetExeVersion)"
-             Condition="'$(OS)' == 'Windows_NT'"/>
-
-    <Exec
-      Condition="!Exists('$(NuGetTool)') And '$(OS)' == 'Windows_NT'"
-      Command="$(NuGetDownloadCommand)"
-      />
-
-    <Exec Command="$(NuGetTool) Pack &quot;$(MSBuildProjectDirectory)\%(NuSpec.Identity)&quot; -verbosity quiet -basepath &quot;$(RootOutDir.TrimEnd('\'))&quot; -outputdirectory &quot;$(NuGetOutDir.TrimEnd('\'))&quot; %(NuSpec.NuGetArguments)" 
-          Condition="'$(OS)' == 'Windows_NT'"/>
+    <!-- This will package the contents of bin/(Configuration)/Raw and create a package in bin/(Configuration)/Packages with Version defined in Common.props -->
+    <MSBuild Projects="$(MSBuildThisFile)"
+             Targets="Pack" 
+             Properties="PackageOutputPath=$(NuGetOutDir.TrimEnd('\'));NoBuild=true;" />
```

Comment:
I think you can just set PackageOutputPath statically in the project, then set GeneratePackageOnBuild to true. We shouldn't need a separate MSBuild invocation and NoBuild isn't necessary since Build of this project is actually required before pack.